### PR TITLE
Implement qPlatform_chmod

### PR DESCRIPTION
**Pull request by fjricci**

### CODE REVIEW

**Comment by a20012251** [comment]
@facebook-github-bot shipit

File: Headers/DebugServer2/GDBRemote/DummySessionDelegateImpl.h [hunk 1]
```diff
@@ -231,6 +231,8 @@ class DummySessionDelegateImpl : public SessionDelegate {
   ErrorCode onFileRemove(Session &session, std::string const &path) override;
   ErrorCode onFileReadLink(Session &session, std::string const &path,
                            std::string &resolved) override;
+  ErrorCode onFileSetPermissions(Session &session, std::string const &path,
```
Comment (sas) [inline Headers/DebugServer2/GDBRemote/DummySessionDelegateImpl.h]: You should put this in the FileOperations mixin.

Comment (sas) [inline Headers/DebugServer2/GDBRemote/DummySessionDelegateImpl.h]: Wait, you're already doing it, scratch that :)

**Comment by sas** [comment]
@facebook-github-bot shipit

File: Sources/GDBRemote/Session.cpp [hunk 2]
```diff
@@ -1957,6 +1958,31 @@ void Session::Handle_qP(ProtocolInterpreter::Handler const &,
...
+//
+void Session::Handle_qPlatform_chmod(ProtocolInterpreter::Handler const &,
+                                     std::string const &args) {
+  char *eptr;
+  uint32_t mode = std::strtoul(args.c_str(), &eptr, 16);
+  if (*eptr++ != ',') {
+    sendError(kErrorInvalidArgument);
+    return;
+  }
+
```
Comment (kokarez) [inline Sources/GDBRemote/Session.cpp]: Are we sure than the string will be long enough ?

if (!*eptr) {
   // error
}

**Pull request merged**